From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Bjarne Koll <git@lynxplay.dev>
Date: Mon, 27 Apr 2026 12:59:01 +0100
Subject: [PATCH] Expose player continuous identifier

Exposes the internally maintained identifier int of a player for plugin
mapping purposes.

diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
index 9a31ac1807e51271c9394026e719d03b67f09f5c..78c23d32f6a015cc5662bf734e14db6a671c77f3 100644
--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
+++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
@@ -3433,4 +3433,13 @@ public class CraftPlayer extends CraftHumanEntity implements Player, PluginMessa
         return com.destroystokyo.paper.profile.CraftPlayerProfile.asBukkitCopy(this.getHandle().originalGameProfile);
     }
     // KTP end - expose original game profile
+    // KTP start - expose continuous identifier to API
+    /**
+     * {@return an integer uniqily identifier this player instance among all other, currently online players.}
+     */
+    @Override
+    public int continuousIdentifier() {
+        return this.getHandle().connection.continuousIdentifier();
+    }
+    // KTP end - expose continuous identifier to API
 }
